lang/java/reef-wake/wake/src/main/java/org/apache/reef/wake/impl/ThreadPoolStage.java

```diff
@@ -206,9 +203,11 @@ public void run() {
 
   /**
    * Closes resources.
+   *
+   * @throws WakeRuntimeException
    */
   @Override
-  public void close() {
+  public void close() throws WakeRuntimeException {
```

Comment:
Please make sure that the `.close()` method never throws (even unchecked exceptions) - as discussed here: [REEF-1729 comment 15906396](https://issues.apache.org/jira/browse/REEF-1729?focusedCommentId=15906396)